packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -906,7 +906,7 @@ export default class Meeting extends StatelessWebexPlugin {
      * @public
      * @memberof Meeting
      */
-    this.recordingController = new RecordingController(this.meetingRequest, this.locusInfo);
+    this.recordingController = new RecordingController(this.meetingRequest);
```

Comment:
question: previously we were setting the locusInfo here, now we're relying on the various LOCUSINFO.EVENTS.* to populate all the things we need. Are these events guaranteed to fire at least once? I mean can there be cases when we got all locus info at the point of meeting object creation and there are no changes to it later so these events don't trigger?

Reply:
This is a good question. I had the same question. I don't think so because the locusInfo is *always* instantiated with "nothing" and then it slowly gets populated with various DTOs, but it's a reasonable thing to build in elegance for.

What i *could* do to solve this problem is instantiate it with the various objects it requires from locusInfo here during the constructor phase and then still leave the event listeners, and that would handle both cases (in case locus changes their implementation I suppose)

As of right now, this code will work based on Locus, but *could* be hardened a bit more.

Reply:
as an example, the *services* links DTO will *always* come in later and therefore must *always* trigger an event, and that's the logic we rely on. I know for sure the fullState is the same. locusUrl I guess they could change at any time?

Reply:
Yeah I think it's best if we just call this.recordingController.set() here with the current values from locusInfo and keep the listeners for any later changes. Then we are covered for all possible cases.

Reply:
From what i have seen in most of the cases the values comes during the join as part of locus (Meeting related property and persmissions )

But those values do change based on what host does 

in some cases it differs if the suer is guest or in lobby so we need to be careful when we test the scenarios out

Reply:
LocusUrl could change if there is a locus migration happening but the locusInfo wont as the meeting remains the same

Reply:
adjusted based on all these comments ! tested and looking good so far